File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> pairs = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            if (pairs.containsKey(target - num)) {
                return new int[] {i, pairs.get(target - num)};
            }
            pairs.put(num, i);
        }
        return new int[] {};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i in range(len(nums)):
            new_target = target - nums[i]
            for j in range(i+1, len(nums)):
                if nums[j] == new_target:
                    return [i, j]